resources/assets/coffee/_classes/beatmap-pack.coffee

```diff
@@ -0,0 +1,72 @@
+###
+#    Copyright 2015-2017 ppy Pty. Ltd.
+#
+#    This file is part of osu!web. osu!web is distributed with the hope of
+#    attracting more community contributions to the core ecosystem of osu!.
+#
+#    osu!web is free software: you can redistribute it and/or modify
+#    it under the terms of the Affero GNU General Public License version 3
+#    as published by the Free Software Foundation.
+#
+#    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+#    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#    See the GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+###
+
+class @BeatmapPack
+  @initialize: ->
+    new BeatmapPack(elem) for elem in document.querySelectorAll('.js-beatmap-pack')
+
+  constructor: (rootElement) ->
+    @el = rootElement
+    @packId = rootElement.dataset.packId
+    @packBody = @el.querySelector('.js-accordion__item-body')
+    @expander = @el.querySelector('.js-accordion__item-header')
+    @busy = false
+    @isCurrent = false
+
+    $.subscribe 'beatmappack:clicked', @onClick
```

Comment:
the rate this is called is quite... phenomenal. Also this needs to be unsubscribed on navigation otherwise it'll be even better when someone go back and forth this page.